utils/__init__.py

```diff
@@ -79,6 +82,45 @@ def push_docker_image_to_repository(
     if username is not None and password is not None:
         docker_push_kwags["auth_config"] = {"username": username, "password": password}
     try:
-        docker_client.images.push(**docker_push_kwags)
+        docker_client.images.push(
```

Comment:
A way to streamline logs from `docker.api.build`:
```python
logs = []
try:
    resp = docker_client.api.build(
        timeout=3600,
        path=".",
        nocache=False,
        buildargs=build_args,
        dockerfile=dockerfile_path,
        tag=image_tag,
    )
    last_event, image_id = None, None
    output = ""
    while True:
        try:
            # output logs to stdout
            # https://docker-py.readthedocs.io/en/stable/user_guides/multiplex.html
            output = next(resp).decode('utf-8')
            json_output = json.loads(output.strip('\r\n'))
            if 'stream' in json_output:
                # output docker build process
                print(json_output['stream'])
                built = re.compile(
                    r'(^Successfully built |sha256:)([0-9a-f]+)$'
                )
                matched = built.search(json_output['stream'])
                if matched:
                    image_id = matched.group(2)
                last_event = json_output['stream']
                logs.append(json_output)
        except StopIteration:
            print(f"Successfully built {image_tag}.")
            break
        except ValueError:
            print(f"Errors while building image:\n{output}", file=sys.stderr)
    if image_id:
        image = docker_client.images.get(image_id)
    else:
        raise docker.errors.BuildError(
            last_event or 'Unknown', logs
        )
except docker.errors.BuildError as e:
    print(f"Failed to build {image_tag} :\n{e.msg}")
    for line in e.build_log:
        if 'stream' in line:
            print(line['stream'].strip())
```

Reply:
wow, this is nice!
should I add it though? currently I was not using that function. using SAM to build and push the image

Reply:
oh you don't have to. Pretty neat when debugging.